### DISCUSSION TITLE
Multiple leftcert in default

### DISCUSSION THREAD

**Discussion by smugcloud** [strongswan/strongswan#1649] [Q&A] [answered]
I am running strongswan 5.9.5 and trying to use multiple server certificates for the `leftcert` parameter.  According to the [ipsec.conf reference](https://wiki.strongswan.org/projects/strongswan/wiki/connsection#leftright-End-Parameters), it appears this is supported, but `ipsec statusall` shows the wrong certificate loaded (even after `ipsec rereadall` and `ipsec reload`).

Is it possible to assign multiple leftcert's in the default section, or do we need to explicitly add them to connections that use something outside of the default?

```
Since 5.0.3 multiple certificate paths or PKCS#11 backends can be specified in a comma separated list.
The daemon chooses the certificate based on the received certificate requests, if possible, before enforcing
the first.
```

ipsec.conf file
```
config setup
...

conn %default
    ike=aes256gcm128-sha512-ecp521
    sp=aes256gcm128-sha512-ecp521
    # sunCert.Pem is the sample cert from https://docs.strongswan.org/docs/5.9/pki/pkiQuickstart.html
    leftcert=sunCert.pem, secure-int.crt 

conn TEST
# uncommenting this configures the proper connection
# leftcert=secure-int.crt
   auto=add
```

ipsec statusall output
```
TEST:  %any...%any  IKEv1/2
TEST:   local:  [C=CH, O=strongswan, CN=sun.strongswan.org] uses public key authentication
TEST:    cert:  "C=CH, O=strongswan, CN=sun.strongswan.org"
TEST:   remote: [CN=TEST] uses public key authentication
TEST:   child:  dynamic === dynamic TUNNEL
```

**Comment by drcyberg** [comment]
Hi!
Yes, it is possible. You must use the `also = <name>` separated for more `conn TEST2`.
You can find the details here: [LINK](https://wiki.strongswan.org/projects/strongswan/wiki/IpsecConf#Reusing-Existing-Parameters)

**Comment by tobiasbrunner** [comment, marked as answer]
> Is it possible to assign multiple leftcert's in the default section, or do we need to explicitly add them to connections that use something outside of the default?

You can assign multiple and if they all contain the configured local identity (e.g. a shared FQDN as SAN) any one could get selected based on the received certificate requests (e.g. to choose between an RSA and an ECDSA PKI). But generally only the first one will be used as the configured local identity has to match, otherwise it defaults to the subject DN of the first certificate anyway and any other certificate will probably not be eligible.

**Comment by smugcloud** [comment]
Thanks for the insight @drcyberg and @tobiasbrunner.  I believe I am running into what you describe @tobiasbrunner, and adding the `also` field may not help us reduce config duplication.  We will add the cert to these other connections.
